server-spi-private/src/main/java/org/keycloak/services/clientpolicy/ClientPolicyProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.services.clientpolicy;
+
+import java.util.List;
+
+import org.keycloak.provider.Provider;
+import org.keycloak.services.clientpolicy.condition.ClientPolicyConditionProvider;
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutorProvider;
+
+/**
+ * Provides Client Policy which accommodates several Conditions and Executors.
+ */
+public interface ClientPolicyProvider extends Provider {
```

Comment:
@tnorimat @stianst Maybe you discuss this already, but I am not 100% sure if ClientPolicyProvider need to be really Java provider? Isn't it rather just a ComponentModel?

Per my understanding, the ClientPolicy provides binding between the set of conditions and executors. Hence usually I the typical administrator will create the policy instance in the admin console. And then he create some conditions and executors and bind them to this policy.

I am probably missing some use-case why the ClientPolicyProvider is beneficial as a java provider rather than as a "model" ?

Reply:
The idea behind this is that there can be multiple policies for a realm, where some are hard-coded/built-in. For example we could have a FAPIClientPolicyProvider that provides a built-in FAPI policy. Not quite sure if this will be the cleanest approach or not, but I would like to somehow avoid that we have to create "built-in" policies when new realms are being created.

Reply:
Perhaps we can keep it like this for now, then when we start adding the FAPI built-in profile, the editor for creating custom policies in the UI, and such, we can consider if it still should be a provider?

Reply:
@stianst +1, that works for me